barretenberg/cpp/src/barretenberg/ecc/groups/affine_element_impl.hpp

```diff
@@ -218,22 +237,58 @@ affine_element<Fq, Fr, T> affine_element<Fq, Fr, T>::hash_to_curve(uint64_t seed
 
     bool y_bit = hash.get_bit(255);
 
-    Fq x_out = Fq(x_coordinate);
-    Fq y_out = (x_out.sqr() * x_out + T::b);
-    if constexpr (T::has_a) {
-        y_out += (x_out * T::a);
+    std::optional<affine_element> result = derive_from_x_coordinate(x_coordinate, y_bit);
+
+    if (result.has_value()) {
+        return result.value();
     }
+    return affine_element(0, 0);
```

Comment:
Why does this return (0,0) instead of nullopt? (I'm assuming that if a point is not on the curve, then the caller will need to check for (0,0)?

Reply:
Coming back to this and other methods which return (0,0) to see how it gets used by callers

Reply:
I refactored the other `hash_to_curve` algo to remove this (0,0) problem. Leaving this one alone because it's getting deleted in the next PR

The reason for the (0, 0) is to backwards compatibility with old code that I don't want to refactor because it's going to get deleted in the next stage of this project.